src/SA_test.jl

```diff
@@ -0,0 +1,39 @@
+using GeometryBasics
+using GeometryBasics.StructArrays
+
+p1 = Point(2, 1)
+p2 = Point(3, 2)
+
+Feature(x; kwargs...) = Feature(x, values(kwargs))
+
+Base.getproperty(x::Feature, s::Symbol) = s == :geometry ? getfield(x, 1) : getproperty(getfield(x, 2), s) # looking for `s` in `properties`
```

Comment:
It's incorrect to define this without also overloading `propertynames`.

Reply:
You mean `getnamestypes` should be `Base.propertynames`?

Reply:
And a separate method for types ?

Reply:
`Base.propertynames` needs to be implemented for `Feature`, this is separate from the `getnamestypes` function, which is just a little helper function that is only used in the StructArrays README example. See the help of `propertynames` for what it should return.